Core/COObject+RelationshipCache.m

```diff
@@ -61,6 +61,12 @@ - (void) removeCachedOutgoingRelationshipsForCollectionValue: (id)obj
 		}
 		else
 		{
+            if (![obj isKindOfClass: [COObject class]])
```

Comment:
I added a comment to explain the purpose of this hack..

This case only occurs during the test suite for NSNull, when we explicitly insert it in the -testNullDisallowedInCollection to check that an exception is thrown. There's nothing special about NSNull though, it could be any ObjC class that we don't support adding to a particular collection.

The problem is, even though an exception is thrown at insertion time, the illegal object still gets inserted. Now that we clean up properly when an object graph is dealloced, this method was running and throwing another exception, when it tried to call `-incomingRelationshipCache` on the collection element. In fact, this issue was breaking UnitKit, because the exception was being thrown when an autorelease pool was drained that wasn't protected by a `@try`.